pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageIdImpl.java

```diff
@@ -90,30 +91,15 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object o) {
-        if (o instanceof MessageIdImpl) {
-            MessageIdImpl other = (MessageIdImpl) o;
-            int batchIndex = (o instanceof BatchMessageIdImpl) ? ((BatchMessageIdImpl) o).batchIndex : NO_BATCH;
-            return messageIdEquals(
-                this.ledgerId, this.entryId, this.partitionIndex, this.batchIndex,
-                other.ledgerId, other.entryId, other.partitionIndex, batchIndex
-            );
-        } else if (o instanceof TopicMessageIdImpl) {
-            return equals(((TopicMessageIdImpl) o).getInnerMessageId());
+        if (o instanceof MessageId) {
+            return compareTo((MessageId) o) == 0;
```

Comment:
This will throw an exception if `o` is an instance of `MessageId` but not `MessageIdImpl` nor `TopicMessageIdImpl`, like `MultiMessageIdImpl`.
Before this PR, it just return false.

Reply:
Updated. I change the impl in `MultiMessageIdImpl` also since it previous throws exception on type mistach in `equals` which seems unnecessary.